### ISSUE TITLE
API for delete_entity_by_id() in specific partition

### ISSUE THREAD

**Issue by hs-pedro** [milvus-io/milvus#4806]
**Is your feature request related to a problem? Please describe.**
The scenario I have is a huge collection (>100M) with over 300 partitions. I wish I could use Milvus to perform deletions of an entity in a single partition (or just in some specific partitions, array of partitions). Currently, if I recall correctly the deletion is by ID in the whole collection.

**Describe the solution you'd like**
It would be awesome if I could pass an array of partitions, similar to other APIs (i.e: search).

**Describe alternatives you've considered**
Currently, on WAL method [DeleteById](https://github.com/milvus-io/milvus/blob/master/core/src/db/wal/WalManager.cpp#L578) the `record` is created with `partition_tag` empty (https://github.com/milvus-io/milvus/blob/master/core/src/db/wal/WalManager.cpp#L588-L591):

```cpp
    MXLogRecord record;
    record.type = MXLogType::Delete;
    record.collection_id = collection_id;
    record.partition_tag = "";
``` 

One alternative, would be to receive an array of partitions looping through them to delete that record from each - or receive a single partition name and set the `record.partition_tag`.

**Comment by yhmo** [comment]
The implementation is similar to #4564

**Comment by shengjun1985** [comment]
@hs-pedro
The modification involves WAL and dbimpl.
Are you able and willing to implement this function for us on branch 1.x?

**Comment by yhmo** [comment]
Implemented in 1.1.0
